core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java

```diff
@@ -140,12 +157,25 @@ public static ProvisioningConfig buildConfig(ProvisioningManager pm,
             final FeaturePackConfig.Builder fpConfig = fp.isTransitive() ? FeaturePackConfig.transitiveBuilder(fpl)
                     : FeaturePackConfig.builder(fpl);
             if (fp.isInheritConfigs() == null) {
-                fpConfig.setInheritConfigs(false);
+                if (hasLayers) {
+                    fpConfig.setInheritConfigs(false);
+                } else {
+                    if (fpWithDefaults) {
+                        fpConfig.setInheritConfigs(true);
+                    }
+                }
```

Comment:
I assume the default here is `false`. I'm just wondering if this should be `fpConfig.setInheritConfigs(!hasLayers && fpWithDefaults)`.